crates/biome_css_analyze/src/assist/source/use_sorted_properties.rs

```diff
@@ -298,52 +319,68 @@ impl RecessOrderMember {
+                .unwrap_or(usize::MAX),
+            None => usize::MAX,
         }
     }
 }
 
+impl PartialOrd for RecessOrderMember {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
 impl Ord for RecessOrderMember {
-    /// First, sort by node kind. Use the order of declaration of the enum [NodeKindOrder].
-    /// Then, sort by property name. Use the index at which the property appears in [crate::order::PROPERTY_ORDER].
-    /// Then, sort by vendor prefix. Use the index at which the prefix appears in [VENDOR_PREFIXES]
-    /// Then, sort by text range.
     fn cmp(&self, other: &Self) -> Ordering {
-        let self_kind = self.kind();
-        let other_kind = other.kind();
-        if self_kind != other_kind {
-            return self_kind.cmp(&other_kind);
+        // First, sort by node kind
+        let kind_cmp = self.kind().cmp(&other.kind());
+        if kind_cmp != Ordering::Equal {
+            return kind_cmp;
         }
 
-        let self_property_index = self.property_index();
-        let other_property_index = other.property_index();
-        if self_property_index != other_property_index {
-            return self_property_index.cmp(&other_property_index);
+        // Then, sort by property index
+        let self_index = self.property_index();
+        let other_index = other.property_index();
+
+        // If either property is unknown, maintain their original order
+        if self_index.is_none() || other_index.is_none() {
+            return Ordering::Equal;
         }
 
-        let self_vendor_prefix = self.vendor_prefix_index();
-        let other_vendor_prefix = other.vendor_prefix_index();
-        if self_vendor_prefix != other_vendor_prefix {
-            return self_vendor_prefix.cmp(&other_vendor_prefix);
+        let prop_cmp = self_index.unwrap().cmp(&other_index.unwrap());
+        if prop_cmp != Ordering::Equal {
+            return prop_cmp;
         }
 
-        self.0.range().cmp(&other.0.range())
-    }
-}
+        // Then, sort by vendor prefix
+        let prefix_cmp = self.prefix_index().cmp(&other.prefix_index());
+        if prefix_cmp != Ordering::Equal {
+            return prefix_cmp;
+        }
 
-impl PartialOrd for RecessOrderMember {
-    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        Some(self.cmp(other))
+        // Finally, sort by the original text to ensure stable sorting
+        let self_text = self
+            .0
+            .as_css_declaration_with_semicolon()
+            .and_then(css_declaration_to_prop_text)
+            .map(|t| t.text().to_lowercase_cow().to_string())
+            .unwrap_or_default();
+        let other_text = other
+            .0
+            .as_css_declaration_with_semicolon()
+            .and_then(css_declaration_to_prop_text)
+            .map(|t| t.text().to_lowercase_cow().to_string())
+            .unwrap_or_default();
+        self_text.cmp(&other_text)
```

Comment:
This base case should keep equal properties in the same order they appear in the source text, but you've changed it to sort them alphabetically, it seems like that might unpredictably shuffle duplicate properties.

You've added a comment that this makes the sort stable. But the previous version was stable, and this version isn't.